clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java

```diff
@@ -264,6 +264,60 @@ public void testReadBytes() {
         this.subTest(buffer);
     }
 
+    @Test
+    public void testFileAsStringSimpleFile() throws IOException {
+        File tempFile = TestUtils.tempFile();
+        try {
+            String testContent = "Test Content";
+            Files.write(tempFile.toPath(), testContent.getBytes());
+            assertEquals(testContent, Utils.readFileAsString(tempFile.getPath()));
+        } finally {
+            Files.deleteIfExists(tempFile.toPath());
+        }
+    }
+
+    /**
+     * Test to read content of named pipe as string.
+     */
+    @Test
+    public void testFileAsStringNamedPipe() throws Exception {
+
+        // Create a temporary name for named pipe
+        Random random = new Random();
+        long n = random.nextLong();
+        n = n == Long.MIN_VALUE ? 0 : Math.abs(n);
+
+        // Use the name to create a FIFO in tmp directory
+        String tmpDir = System.getProperty("java.io.tmpdir");
+        String fifoName = "fifo-" + n + ".tmp";
+        File fifo = new File(tmpDir, fifoName);
+        Thread producerThread = null;
+        try {
+            Process mkFifoCommand = new ProcessBuilder("mkfifo", fifo.getCanonicalPath()).start();
```

Comment:
Is this command guaranteed to be available in every system? We may want to make it conditional if not. Or perhaps there's another way to test this that doesn't involve using an external command?

Reply:
This is part of `Gnu coreutils`. So if you can do things like `ls`, `cat` or `echo`, you should be able to run `mkfifo` too. 

Here are list of coreutils command: https://en.wikipedia.org/wiki/List_of_GNU_Core_Utilities_commands